platform/default/headless_view.cpp

```diff
@@ -32,26 +38,41 @@ HeadlessView::~HeadlessView() {
 }
 
 void HeadlessView::resize(const uint16_t width, const uint16_t height) {
-    if(dimensions[0] == width &&
-       dimensions[1] == height) {
+    if (dimensions[0] == width && dimensions[1] == height) {
+        return;
+    }
+
+    if (width > dimensionsLimits[0] || height > dimensionsLimits[1]) {
         return;
     }
+
     dimensions = {{ width, height }};
     needsResize = true;
 }
 
-PremultipliedImage HeadlessView::readStillImage() {
+std::shared_ptr<const PremultipliedImage> HeadlessView::readStillImage() {
+    MBGL_VERIFY_THREAD(tid);
+
     assert(active);
+    assert(dimensions[0] <= dimensionsLimits[0]);
+    assert(dimensions[1] <= dimensionsLimits[1]);
+
+    static auto image = std::make_shared<PremultipliedImage>(
```

Comment:
Ack, didn't catch that - this doesn't seem workable given how we have have up to 8 `NodeMap` objects per `node` process currently, each with their own `HeadlessView`?